### PR TITLE
Fix issue 142

### PR DESCRIPTION
**Pull request by fteicht**
Fix issue #142

### CODE REVIEW

File: cpp/src/hub/solver/mcts/mcts.hh [hunk 1]
```diff
@@ -97,17 +97,11 @@ public:
       HeuristicFunctor;
 
   FullExpand(
-      const HeuristicFunctor &heuristic =
+      const HeuristicFunctor &heuristic =
```
Comment (dbarbier) [inline cpp/src/hub/solver/mcts/mcts.hh:100]: whitespace changes are bad (and break linters by the way)

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -18,6 +18,10 @@ IF (NOT WIN32 AND NOT CYGWIN)
     SET(CMAKE_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden ${CMAKE_CXX_FLAGS}")
 ENDIF ()
 
+IF (MSVC)
+    ADD_COMPILE_OPTIONS(/MP /Zc:lambda)
+ENDIF ()
```
[the remark this answers is not in the document]
Comment (dbarbier) [reply cpp/CMakeLists.txt:23]: Hmmm it looks like the 3rd item was wrong, MSVC silently ignores unknown flags.  But this is not true for clang, so my recommendation still stands.

**Comment by dbarbier** [comment]
@fteicht Your changes break windows-2016 builds; any chance to have a version which works with any windows-* version?

File: cpp/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (fteicht) [reply cpp/CMakeLists.txt:23]: Actually older versions of MSVC silently ignore the lambda compilation flag but they cannot parse the simplified version of MCTS' FullExpand's constructor (another independent issue). So I will revert the corresponding commit.

**Comment by fteicht** [comment]
> 

Yes because I've just figured out that old versions of MSVC (e.g. 2017 and before) cannot correctly parse typedef members inside lambda functions. It was corrected by the introduction of the `Zc:lambda` compilation flag but it only works with newest versions of MSVC. Thus I've reverted the commit that simplifies MCTS' FullExpand's constructor implementation.

File: cpp/src/hub/solver/riw/riw.hh [hunk 3]
```diff
@@ -215,6 +215,13 @@ private:
     typedef Trollout_policy<Domain> RolloutPolicy;
     typedef Texecution_policy ExecutionPolicy;
 
+  private:
+    typedef typename ExecutionPolicy::template atomic<std::size_t>
+        atomic_size_t;
+    typedef typename ExecutionPolicy::template atomic<double> atomic_double;
+    typedef typename ExecutionPolicy::template atomic<bool> atomic_bool;
```
Comment (dbarbier) [inline cpp/src/hub/solver/riw/riw.hh:222]: These typedefs appear in public methods, why are they private?

**Review by dbarbier — APPROVED** [review]
Well done, looks good to me.

Since windows-2016 Github runners are going to be removed on March 2022, you should replace windows-2016 by windows-latest on l.40 of build.yml 
          build = [ "macos-10.15", "ubuntu-latest", "windows-2016" ]
and l. 29 of release.yml
          build_dict = { "macos":["macos-10.15"], "ubuntu":["ubuntu-latest"], "windows":["windows-2016"] }
On Windows backward compatibility is ensured, so we can build on -latest.

And bonus points if you squash your commits ;-)

File: cpp/src/hub/solver/riw/riw.hh (hunk 3 above)
Comment (fteicht) [reply cpp/src/hub/solver/riw/riw.hh:221]: You're right, I've just committed the change. Thanks!

**Comment by dbarbier** [comment]
@fteicht  Please remove the merge commit, it does not provide any value:

```
$ git diff 45a5132 7bf5f70
$
```

You can run
```
 git reset --hard 45a5132
 git push --force-with-lease origin fix-issue-142
 ```

**Pull request merged**